### PR TITLE
Add more caliper annotations and makes random uq parallel in devices

### PR DESCRIPTION
**Pull request by koparasy**

### CODE REVIEW

File: src/AMSlib/ml/random_uq.hpp [hunk 1]
```diff
@@ -12,15 +12,23 @@
...
 {
 public:
   PERFFASPECT()
-  inline void evaluate(const size_t ndata, bool *is_acceptable) const
+  inline void evaluate(const size_t ndata, bool *is_acceptable)
   {
     if (resourceLocation == AMSResourceType::DEVICE) {
 #ifdef __ENABLE_CUDA__
-      random_uq_device<<<1, 1>>>(is_acceptable, ndata, threshold);
+      //TODO: Move all of this code on device.cpp and provide
```
Comment (ggeorgakoudis) [inline src/AMSlib/ml/random_uq.hpp]: Comment seems unfinished

File: src/AMSlib/ml/uq.hpp [hunk 2]
```diff
@@ -17,9 +17,7 @@
 #include "ml/surrogate.hpp"
 #include "wf/resource_manager.hpp"
 
-static inline bool isNullOrEmpty(const char *p) {
-  return (!p || p[0] =='\0');
-}
+static inline bool isNullOrEmpty(const char *p) { return (!p || p[0] == '\0'); }
```
Comment (ggeorgakoudis) [inline src/AMSlib/ml/uq.hpp:20]: Format is off

File: src/AMSlib/ml/uq.hpp [hunk 3]
```diff
@@ -33,12 +31,17 @@ class UQ
      FPTypeValue threshold)
       : uqPolicy(uqPolicy), threshold(threshold)
   {
+    if (isNullOrEmpty(surrogatePath)) {
```
Comment (ggeorgakoudis) [inline src/AMSlib/ml/uq.hpp:34]: Do we allow the absence of a surrogate? What happens in this case?

Comment (koparasy) [reply src/AMSlib/ml/uq.hpp:34]: It falls backs to the original existing physics code and performs data collection.

File: src/AMSlib/ml/uq.hpp (hunk 2 above)
Comment (koparasy) [reply src/AMSlib/ml/uq.hpp:20]: `clang-format --style=file -i src/AMSlib/ml/uq.hpp` makes this.

**Review by ggeorgakoudis — APPROVED** [review]

**Pull request merged**